### ISSUE TITLE
what type should be passed to `Data.streamfrom` for `Data.Column`?

### ISSUE THREAD

**Issue by ExpandingMan** [JuliaData/DataStreams.jl#31]
It seems to me that the type that gets passed to `Data.streamfrom(source, Data.Column, ...)` isn't standardized properly yet.  In some cases it seems to be the container type, in others it is the element type.

In `DataTables`, one can do either since there is an extra `Data.streamfrom` method defined that doesn't use the type parameter at all.  In the documentation, it says that the type passed should be the element type.

In `Feather.jl` as well as for some of the methods in `DataTables` it seems to require the container type.  Right now it seems like more often the container type is needed, so perhaps the documentation should be changed.

I think it makes more sense to pass the container type, as ideally the work of packing the data into it should be done by the source, not by whatever is pulling from it.

**Comment by quinnj** [comment]
It depends on the `StreamType`. If a `Source` is streaming `Field`s, the types will be eltypes, if it's streaming `Column`s, the types will be container types. This should probably be clarified in the docs.

**Comment by ExpandingMan** [comment]
That was pretty much what I inferred, but what I was saying was that sometimes it seemed like even `Data.Column` should get the eltype, not the container type (in the documentation and looking at the `DataTables` API).  It might be nice to take another pass through the documentation.

When you're doing that, you might want to consider #30, I've already encountered a number of cases where it is relevant.

Anyway, thanks for the clarification.
